src/NHibernate/NHibernateUtil.cs

```diff
@@ -33,10 +33,11 @@ static NHibernateUtil()
 				clrTypeToNHibernateType[type.ReturnedClass] = type;
 			}
 
-			// There are multiple possibilites for boolean and strings.
+			// There are multiple possibilites for boolean, strings and datetime.
 			// Override so that we use the most natural mapping.
 			clrTypeToNHibernateType[Boolean.ReturnedClass] = Boolean;
 			clrTypeToNHibernateType[String.ReturnedClass] = String;
+			clrTypeToNHibernateType[DateTime.ReturnedClass] = DateTime;
```

Comment:
In case #702 does not get merge, at least fix NHibernateUtil.GuesType yielding a random type for `DateTime`.